### PR TITLE
More optimizations of the geopotential

### PR DESCRIPTION
**Pull request by pleroy**
Pass in quantities computed by the ephemeris and use uninitialized arrays.  Before:
```
Run on (4 X 3310 MHz CPU s)
10/02/18 23:43:21
----------------------------------------------------------------------------------------
Benchmark                                                 Time           CPU Iterations
----------------------------------------------------------------------------------------
BM_EphemerisSolarSystemMajorBodiesOnly/-3        46441258970 ns 45723893100 ns          1 +9.92179030813603147e-01 ua
BM_EphemerisSolarSystemMinorAndMajorBodies/-3    113307575563 ns 111337913700 ns          1 +9.92179031500272979e-01 ua
BM_EphemerisSolarSystemAllBodiesAndOblateness/-3 286661034049 ns 284608224400 ns          1 +9.92178865021134948e-01 ua
```
After:
```
Run on (4 X 3310 MHz CPU s)
10/06/18 11:36:42
----------------------------------------------------------------------------------------
Benchmark                                                 Time           CPU Iterations
----------------------------------------------------------------------------------------
BM_EphemerisSolarSystemMajorBodiesOnly/-3        44335378231 ns 44304284000 ns          1 +9.92179030813607254e-01 ua
BM_EphemerisSolarSystemMinorAndMajorBodies/-3    107144056865 ns 107047886200 ns          1 +9.92179031500228903e-01 ua
BM_EphemerisSolarSystemAllBodiesAndOblateness/-3 249901346662 ns 242191552500 ns          1 +9.92178865021122292e-01 ua
```
After after:
```
Run on (4 X 3310 MHz CPU s)
10/13/18 14:06:47
----------------------------------------------------------------------------------------
Benchmark                                                 Time           CPU Iterations
----------------------------------------------------------------------------------------
BM_EphemerisSolarSystemMajorBodiesOnly/-3        43925861411 ns 43883081300 ns          1 +9.92179030813603147e-01 ua
BM_EphemerisSolarSystemMinorAndMajorBodies/-3    107695463871 ns 107266287600 ns          1 +9.92179031500272979e-01 ua
BM_EphemerisSolarSystemAllBodiesAndOblateness/-3 237049283971 ns 234391502500 ns          1 +9.92178865021101086e-01 ua
```

### CODE REVIEW

File: physics/geopotential_body.hpp [hunk 1]
```diff
@@ -50,17 +51,17 @@ struct Geopotential<Frame>::Precomputations {
   Vector<Inverse<Length>, Frame> grad_𝔏_vector;
 
   // These quantities depend on n but are independent from m.
-  FixedVector<Inverse<Length>, size> ℜ;  // 0 unused.
+  FixedVector<Inverse<Length>, size> ℜ{uninitialized};  // 0 unused.
```
Comment (eggrobin) [inline physics/geopotential_body.hpp]: Here uninitialized does nothing unless you add an uninitialized constructor to Quantity.

We should do that.

I could do that in a separate pull request; add a TODO.

Comment (pleroy) [reply physics/geopotential_body.hpp]: #1958

File: astronomy/geodesy_test.cpp [hunk 2]
```diff
@@ -267,16 +267,16 @@ TEST_F(GeodesyTest, LAGEOS2) {
   // Absolute error in position.
   EXPECT_THAT(AbsoluteError(secondary_actual_final_dof.position(),
                             primary_actual_final_dof.position()),
-              IsNear(23 * Metre));
+              IsNear(560 * Metre));
   // Angular error at the geocentre.
   EXPECT_THAT(AngleBetween(secondary_actual_final_dof.position() - ITRS::origin,
                            primary_actual_final_dof.position() - ITRS::origin),
-              IsNear(388 * Milli(ArcSecond)));
+              IsNear(9367 * Milli(ArcSecond)));
```
[the remark this answers is not in the document]
Comment (pleroy) [reply astronomy/geodesy_test.cpp]: Done.
